lib/addPipelineExtras.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+
+module.exports = addPipelineExtras;
+
+//Add the pipeline extras object to each glTF object.
+function addPipelineExtras(gltf) {
+    var reference = {

```

Comment:
After looking over it again, I think the main reason I had the `addPipelineExtra` property and the additional reference objects like `accessorObject`, `animationObject`, etc. in the reference was to differentiate between glTF objects which directly had extras, like:

```
"asset": {
        "generator": "collada2gltf@ceec062e3d5793f2f249f53cbd843aee382ad40b",
        "premultipliedAlpha": true,
        "profile": {
            "api": "WebGL",
            "version": "1.0.2"
        },
        "version": 1
    },
```
where `extras` is added to the `asset` object,
vs. 
```
"accessors": {
        "accessor_21": {
            "bufferView": "bufferView_29",
            "byteOffset": 0,
            "componentType": 5123,
            "count": 36,
            "type": "SCALAR"
        }
    }
```
where `extras` is added to the `accessor_21` object.
So, I think I would still need some kind of marker variable which determines which kind of glTF object it is.